javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -783,7 +824,51 @@ public boolean isTraversableAgain() {
 
     @Override
     public Iterator<T> iterator() {
-        return isEmpty() ? Iterator.empty() : trie.iterator();
+        return isEmpty() ? Iterator.empty()
+                         : new VectorIterator();
+    }
+
```

Comment:
Thanks, but I did test it and it was surprisingly slow.
Will try to come up with something better, if you don't like it.